src/backend/login/usingPassport.js

```diff
@@ -1,36 +1,40 @@
-const passport = require('passport');
-const OneLoginStrategy = require('passport-openid').Strategy;
+const saml = require('passport-saml');
 require('../lib/config');
+const fs = require('fs');
+const path = require('path');
 
-/* Set environmental variables for these
-https://www.onelogin.com/ to create your account.
-Tutorial is here: https://developers.onelogin.com/quickstart/authentication/nodejs
-OIDC_BASE_URI=https://openid-connect.onelogin.com/oidc
-OIDC_CLIENT_ID=ddf97d-xxx-xxxx-e92920
-OIDC_CLIENT_SECRET=f021f63bxxxx71f2c6a90c9b
-OIDC_REDIRECT_URI=http://localhost:3000/oauth/callback
+/*
+Tutorial is here: https://medium.com/disney-streaming/setup-a-single-sign-on-saml-test-environment-with-docker-and-nodejs-c53fc1a984c9
+SAML2_BASE_URI=saml-poc
+SAML2_CLIENT_ID=ddf97d-xxx-xxxx-e92920
+SAML2_CLIENT_SECRET=f021f63bxxxx71f2c6a90c9b
+SAML2_REDIRECT_URI=http://localhost:3000/oauth/callback
 */
 
-// Configure the OpenId Connect Strategy
-// with credentials obtained from OneLogin
-const { OIDC_BASE_URI, OIDC_CLIENT_ID, OIDC_CLIENT_SECRET, OIDC_REDIRECT_URI } = process.env;
+const { SAML2_BASE_URI, SAML2_REDIRECT_URI } = process.env;
 
-passport.use(
-  new OneLoginStrategy(
-    {
-      issuer: OIDC_BASE_URI,
-      clientID: OIDC_CLIENT_ID,
-      clientSecret: OIDC_CLIENT_SECRET,
-      authorizationURL: `${OIDC_BASE_URI}/auth`,
-      userInfoURL: `${OIDC_BASE_URI}/me`,
-      tokenURL: `${OIDC_BASE_URI}/token`,
-      callbackURL: OIDC_REDIRECT_URI,
-      passReqToCallback: true,
-    },
-    // This function returns 'issuer:' issuer, 'userId:' userId, 'accessToken:' accessToken, 'refreshToken:' refreshToken, 'params:' params
-    (req, issuer, userId, profile, accessToken, refreshToken, params, callback) => {
-      req.session.accessToken = accessToken;
-      return callback(null, profile);
-    }
-  )
+const samlStrategy = new saml.Strategy(
+  {
+    callbackUrl: SAML2_REDIRECT_URI,
+    entryPoint: SAML2_BASE_URI,
+    issuer: 'saml-poc',
+    identifierFormat: null,
+    decryptionPvk: fs.readFileSync(path.resolve(process.cwd(), './certs/key.pem'), 'utf8'),
+    privateCert: fs.readFileSync(path.resolve(process.cwd(), './certs/key.pem'), 'utf8'),
+  },
+  function(profile, done) {
+    return done(null, profile);
+  }
 );
+
+const handleLogin = function(req, res, next) {
+  next();
+};
+// Configure the Saml Strategy
+// with credentials obtained from Saml2 Docker
+// eslint-disable-next-line no-unused-expressions
```

Comment:
Can we avoid these?

```js
module.exports = {
  samlStrategy: new saml.Strategy(...),
  handleLogin,
};
```

Reply:
Sorry, are you referring to the comments or the code underneath to avoid?

Reply:
eslint disables, I'd prefer to not do that.

Reply:
I removed the line I believe now in the latest commit.